File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectBasicsJava/app/src/main/java/pl/edu/uwr/pum/interfaceinjectbasicsjava/di/modules/AmdModule.java
package pl.edu.uwr.pum.interfaceinjectbasicsjava.di.modules;

import dagger.Binds;
import dagger.Module;
import pl.edu.uwr.pum.interfaceinjectbasicsjava.computer.component.AMD;
import pl.edu.uwr.pum.interfaceinjectbasicsjava.computer.component.CPU;

@Module
abstract public class AmdModule {
    @Binds
    abstract CPU provideCpu(AMD cpu);
}

File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectiBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/interfaceinjectibasicskotlin/di/modules/AmdModule.kt
package pl.edu.uwr.pum.interfaceinjectibasicskotlin.di.modules

import dagger.Binds
import dagger.Module
import dagger.Provides
import pl.edu.uwr.pum.interfaceinjectibasicskotlin.computer.components.AMD
import pl.edu.uwr.pum.interfaceinjectibasicskotlin.computer.components.CPU

@Module
abstract class AmdModule {
    @Binds
    abstract fun bindCpu(cpu: AMD): CPU
}